src/Loggers/Loggers.h

```diff
@@ -7,7 +7,7 @@
 #include <Poco/Util/Application.h>
 #include "OwnSplitChannel.h"
 
-#ifdef WITH_TEXT_LOG
+#ifndef WITHOUT_TEXT_LOG
```

Comment:
This now leads `Loggers` to be expected with text log in clickhouse-keeper, but linked with `loggers_no_text_log`. The problem is not fixed.

Please, make a separate PR with typo fixes, because it's hard to read this PR and to understand what is the actual fix.

Reply:
src/CMakeList.txt 319~320
dbms_target_include_directories (PUBLIC "${ClickHouse_SOURCE_DIR}/src" "${ClickHouse_BINARY_DIR}/src")
target_include_directories (clickhouse_common_io PUBLIC "${ClickHouse_SOURCE_DIR}/src" "${ClickHouse_BINARY_DIR}/src")

this two lines expose all .h （include Loggers.h）to  all build libraries (common-io and all dmbs)。during all these libraries building they just not see 'WITH_TEXT_LOG' compile definition。HOW we could make sure some library will 
inherit the compile definition from the loggers (or loggers_no_text_log) library? it must be explicit link loggers (modern cmake target link rules), in the project there only two libraries do this:
target_link_libraries (daemon PUBLIC loggers)
target_link_libraries(clickhouse-keeper PRIVATE loggers_no_text_log)

At last clickhouse-server links daemon library (with `WITH_TEXT_LOG`) and other comon_io / dbms libraries (without `WITH_TEXT_LOG`), the Loggers class definition conflicts.

Above is my explanation. as for the mini-clickhouse-keeper without text log, its link dependencies have no change,  will keep small. But may still have problem, becase clickhouse-keeper links loggers_no_text_log and clickhouse_common_io, the compile definition (`WITH_TEXT_LOG`) in loggers_no_text_log not exposed to clickhouse_common_io.
target_link_libraries(clickhouse-keeper
        PRIVATE
           ...
            loggers_no_text_log
            clickhouse_common_io
    )

At last，it seems no easy way to fix this problem gracefully  now, just the whole compile target dependency is not struct in modern CMake way https://gist.github.com/mbinna/c61dbb39bca0e4fb7d1f73b0d66a4fd1

Reply:
i make a separate PR for typo fix here, https://github.com/ClickHouse/ClickHouse/pull/40984